src/main/java/com/alvarium/DefaultSdk.java

```diff
@@ -50,7 +50,21 @@ public void create(PropertyBag properties, byte[] data) throws AnnotatorExceptio
   public void mutate(PropertyBag properties, byte[] oldData, byte[] newData) {
   }
 
-  public void transit(PropertyBag properties, byte[] data) {
+  public void transit(PropertyBag properties, byte[] data) throws AnnotatorException,
+      StreamException {
+    final List<Annotation> annotations = new ArrayList<Annotation>();
+    final String contentType = "AnnotationList";
+
+    // Annotate incoming data
+    for (Annotator annotator: this.annotators) {
+      final Annotation annotation = annotator.execute(properties, data);
+      annotations.add(annotation);
+    }
+
+    // publish list of annotations to the StreamProvider
+    final PublishWrapper wrapper = new PublishWrapper(SdkAction.TRANSIT, contentType, annotations);
+    this.stream.publish(wrapper);
+    this.logger.debug("data annotated and published successfully.");
```

Comment:
Go has a different opinion about code duplication than Java and I'm getting complaints in my IDE about the similarities of `create()` and `transit()`.

Some refactoring will need to be done in this area.

Reply:
We can breakdown duplicate functionalities into smaller private methods which we can use freely inside the sdk. We will consider that in a future PR.